tfhe/src/shortint/ciphertext/mod.rs

```diff
@@ -8,11 +8,49 @@ use std::cmp;
 use std::fmt::Debug;
 
 use super::parameters::{CiphertextConformanceParams, CiphertextListConformanceParams};
+use super::CheckError;
 
 /// This tracks the number of operations that has been done.
 #[derive(Debug, PartialEq, Eq, Copy, Clone, Serialize, Deserialize)]
 pub struct Degree(pub usize);
 
+/// This tracks the maximal amount of noise of a [Ciphertext]
+/// that guarantees the target p-error when doing a PBS on it
+#[derive(Debug, PartialEq, Eq, Copy, Clone, Serialize, Deserialize)]
+pub struct MaxNoiseLevel(usize);
+
+impl MaxNoiseLevel {
+    pub fn new(value: usize) -> Self {
+        Self(value)
+    }
+
+    pub fn get(&self) -> usize {
+        self.0
+    }
+
+    // This function is valid for current parameters as they guarantee the p-error for a norm2 noise
+    // limit equal to the norm2 limit which guarantees a clean padding bit
+    //
+    // TODO: remove this functions once noise norm2 constraint is decorrelated and stored in
+    // parameter sets
```

Comment:
if we don't have an issue for this let's create one and assign yourself on it for now

Reply:
https://github.com/zama-ai/tfhe-rs-internal/issues/339